.github/actions/inject-issue-id-action/action.yaml

```diff
@@ -0,0 +1,99 @@
+---
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: 2024 The Linux Foundation
+
+name: "🎟️ Add JIRA ticket to commit message"
+
+inputs:
+  ISSUE_ID_LOOKUP_JSON:
+    description: "JSON mapping users to JIRA tickets"
+    type: string
+    required: true
+  INJECT:
+    # When injection is enabled and no issue found...
+    # String will be added, output variable set true
+    description: "Whether to inject an issue-id if NOT found"
+    type: boolean
+    required: false
+    default: true
+
+outputs:
+  PRESENT:
+    description: "Whether the required ticket string is present"
+    value: ${{ steps.status.outputs.present }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.head_ref }}
+        fetch-depth: 0
```

Comment:
Why are you forcing a complete checkout? Given that this is expected to only be used on PRs and not Gerrit changes, this should really be using the default of 1 for faster checkouts.

Reply:
I was trying to address the issues around the force push and I think I misunderstood what was happening during the checkout step. "github.head_ref" was the required fix and the fetch depth wasn't required. I've updated the checkout step accordingly, and tested again today in the workflow run below:
https://github.com/ModeSevenIndustrialSolutions/test-issue-id/actions/runs/12120595981/job/33789781183